src/utils/swapErrorToUserReadableMessage.tsx

```diff
@@ -22,6 +22,17 @@ export function swapErrorToUserReadableMessage(error: any): string {
     error = error.error ?? error.data?.originalError
   }
 
+  // The 4001 error code doesn't capture the case where users reject a transaction for all wallets,
+  // so we need to parse the reason for these special cases:
+  if (
+    // For rainbow wallet:
+    (reason?.toLowerCase().includes('request') && reason?.toLowerCase().includes('reject')) ||
```

Comment:
Could you simplify this by using regex? Do you know the ordering of `request` and `reject`?

    if (
      reason?.match(/request/i) && reason.match(/reject/i) ||
      reason.match(/declined/i)
    ) {

Reply:
we want to handle the case where request appears before reject and vice versa as both occur (see here: https://uniswapteam.slack.com/archives/C047U65H422/p1675798801630839). i don't think regex is necessary for what we're trying to do here for the sake of readability / simplicity ?

Reply:
I'd prefer regex - I think it is simpler because it omits toLowerCase and is still very clear in what it does (`/i` means case-insensitive) - but this is not a blocker.

Reply:
sg that makes sense done!